pkg/kubelet/hyper/hyper.go

```diff
@@ -0,0 +1,745 @@
+	if len(parts) != 4 {
+		return "", "", "", fmt.Errorf("failed to parse the pod full name %q", podFullName)
+	}
+	return parts[1], parts[2], parts[3], nil
+}
+
+func (r *runtime) parseHyperContainerFullName(containerName string) (string, string, string, string, error) {
+	parts := strings.Split(containerName, "_")
+	if len(parts) != 6 {
+		return "", "", "", "", fmt.Errorf("failed to parse the container full name %q", containerName)
+	}
+	return parts[1], parts[2], parts[3], parts[4], nil
+}
+
+// GetPods returns a list containers group by pods. The boolean parameter
+// specifies whether the runtime returns all containers including those already
+// exited and dead containers (used for garbage collection).
+func (r *runtime) GetPods(all bool) ([]*kubecontainer.Pod, error) {
+	podInfos, err := r.hyperClient.ListPods()
+	if err != nil {
+		return nil, err
+	}
+
+	var kubepods []*kubecontainer.Pod
+	for _, podInfo := range podInfos {
+		if !all && podInfo.status != "running" {
+			continue
+		}
+
+		var pod kubecontainer.Pod
+		var containers []*kubecontainer.Container
+
+		podID, podName, podNamespace, err := r.parseHyperPodFullName(podInfo.podName)
+		if err != nil {
+			glog.Errorf("Hyper: can't parse pod name %s", podInfo.podName)
+			return nil, err
+		}
+
+		pod.ID = types.UID(podID)
+		pod.Name = podName
+		pod.Namespace = podNamespace
+
+		for _, cinfo := range podInfo.podInfo.Spec.Containers {
+			var container kubecontainer.Container
+			container.ID = types.UID(cinfo.ContainerID)
+			container.Image = cinfo.Image
+
+			for _, cstatus := range podInfo.podInfo.Status.Status {
+				if cstatus.ContainerID == cinfo.ContainerID {
+					createAt, err := parseTimeString(cstatus.Running.StartedAt)
+					if err == nil {
+						container.Created = createAt.Unix()
+					}
+				}
+			}
+
+			_, _, _, containerName, err := r.parseHyperContainerFullName(cinfo.Name)
+			if err != nil {
+				return nil, err
+			}
+			container.Name = strings.Split(containerName, ".")[0]
+
+			hash, err := strconv.ParseUint(strings.Split(containerName, ".")[1], 16, 8)
+			if err == nil {
+				container.Hash = hash
+			}
+
+			containers = append(containers, &container)
+		}
+		pod.Containers = containers
+
+		kubepods = append(kubepods, &pod)
+	}
+
+	return kubepods, nil
+}
+
+func (r *runtime) buildHyperPod(pod *api.Pod, pullSecrets []api.Secret) ([]byte, error) {
+	// check and pull image
+	for _, c := range pod.Spec.Containers {
+		if err := r.imagePuller.PullImage(pod, &c, pullSecrets); err != nil {
+			return nil, err
+		}
+	}
+
+	// build hyper volume spec
+	specMap := make(map[string]interface{})
+	volumeMap, ok := r.volumeGetter.GetVolumes(pod.UID)
+	if !ok {
+		return nil, fmt.Errorf("cannot get the volumes for pod %q", kubecontainer.GetPodFullName(pod))
+	}
+
+	volumes := []map[string]string{}
+	for name, volume := range volumeMap {
+		glog.V(4).Infof("Hyper: volume %s %s", name, volume.GetPath())
+		v := make(map[string]string)
+		v["name"] = name
```

Comment:
Nit: might be good to make `"name"`,`"source"`, etc constants to avoid magic strings all over.